src/lib.rs

```diff
@@ -251,6 +251,14 @@ struct State {
     alpha: f32,
 }
 
+impl State {
+    fn with_pixel_ratio(pixel_ratio: f32) -> Self {
```

Comment:
Perhaps the `Default` impl should be removed for `State` to make sure to catch all cases where `State` is re-initialized and a decision should be made whether to apply the pixel ratio or not. It could be `new()` and `new_with_pixel_ratio` perhaps?